api/v1beta1/common.go

```diff
@@ -102,3 +104,25 @@ type NetworksSection struct {
 	// FixedIP - Specific IP address to use for this network
 	FixedIP string `json:"fixedIP,omitempty" yaml:"fixedIP,omitempty"`
 }
+
+// UniqueSpecFields - the array of fields that must be unique between role and nodes
+var UniqueSpecFields = []string{"OpenStackAnsibleEERunnerImage", "NetworkAttachments"}
+
+// AssertUniquenessBetween - compare specs for uniqueness
+func AssertUniquenessBetween(spec interface{}, otherSpec interface{}, suffix string) []string {
```

Comment:
This refactoring makes it cleaner IMO. Thanks!

Reply:
this can be moved into lib-common later on